openedx/core/djangoapps/programs/utils.py

```diff
@@ -150,8 +195,14 @@ def progress(self, programs=None, count_only=True):
             for course in program['courses']:
                 if self._is_course_complete(course):
                     completed.append(course)
-                elif self._is_course_in_progress(course):
-                    in_progress.append(course)
+                elif self._is_course_enrolled(course):
+                    course_in_progress = self._is_course_in_progress(now, enrolled_run_modes, course)
+                    if course_in_progress:
+                        in_progress.append(course)
+                    else:
+                        course_copy = deepcopy(course)
```

Comment:
is it really worth it to do the extra work just to be consistent?  
presumably it is clear that the reason for the copy here is that this course is modified, we could add a comment if that helps